extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -414,20 +417,23 @@ class LifecycleTask extends BackbeatTask {
      * For all filtered rules, get rules that apply the earliest
      * @param {array} rules - list of filtered rules that apply to a specific
      *   object, version, or upload
+     * @param {object} metadata - metadata about the object to transition
      * @return {object} all applicable rules with earliest dates of action
      *  i.e. { Expiration: { Date: <DateObject>, Days: 10 },
      *         NoncurrentVersionExpiration: { NoncurrentDays: 5 } }
      */
-    _getApplicableRules(rules) {
+    _getApplicableRules(rules, metadata) {
         // NOTE: Ask Team
         // Assumes if for example a rule defines expiration and transition
         // and if backbeat disables expiration and enables transition,
         // we still consider this rules transition to apply
-
         // these are rules enabled in config.json
         const enabledRules = Object.keys(this.enabledRules).filter(rule =>
             this.enabledRules[rule].enabled);
 
+        // Declare the current date before the reducing function so that all
+        // rule comparisons use the same date.
```

Comment:
The following comment `//NOTE: only care about expiration...` at line 440 should be removed now.